App auth method policy - trusted CA issuer docs (#21187)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/certificateauthorityasentity-delete.md

---
title: "Delete certificateAuthorityAsEntity"
description: "Delete a certificateAuthorityAsEntity object."
author: "madansr7"
ms.localizationpriority: medium
ms.prod: "identity-and-sign-in"
doc_type: apiPageType
---

# Delete certificateAuthorityAsEntity
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> Delete a [certificateAuthorityAsEntity](../resources/certificateauthorityasentity.md) object.
> 
> >**Note**: You can't delete all items in the collection, because this collection requires at least one object to persist at all times.

Reviewer: Also, can you delete a root authority before designating another object as the root authority? If yes, how does Azure AD auto-assign which object becomes the root authority? Document what answers these questions.
```suggestion
Delete a [certificateAuthorityAsEntity](../resources/certificateauthorityasentity.md) object. You can't delete all items in the collection because this collection requires at least one object that is a root authority to always persist.
```

> ---

Reviewer 2: Please, include an update to the toc.yml file in the BETA folder, to surface all new APIs in the TOC.

Author: done.

title: "Delete certificateAuthorityAsEntity"
description: "Delete from the trustedCertificateAuthorities collection."
author: "madansr7"
ms.localizationpriority: medium
ms.prod: "identity-and-sign-in"
doc_type: apiPageType
---

# Delete certificateAuthorityAsEntity
Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

Delete from the trustedCertificateAuthorities collection.

>**Note**: You can't delete all items in the collection, because this collection requires at least one object to persist at all times.

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type|Permissions (from least to most privileged)|
|:---|:---|

> |Delegated (work or school account)|AppCertTrustConfiguration.Read.All, AppCertTrustConfiguration.ReadWrite.All|

Reviewer 2: Your PR introduces new permissions. Please list these new permissions in the https://learn.microsoft.com/en-us/graph/permissions-reference topic.

Author: added. thanks for catching it.
Is this something the doc stub tool should also do? I probably missed it because i copied all files from the stub gen tool.

Reviewer 2: This should be added manually.

|Delegated (personal Microsoft account)|AppCertTrustConfiguration.Read.All, AppCertTrustConfiguration.ReadWrite.All|
|Application|None|

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->
``` http
DELETE /directory/certificateAuthorities/certificateBasedApplicationConfigurations/{certificateBasedApplicationConfigurations-Id}/trustedCertificateAuthorities/{trustedCertificateAuthorities-Id}
```

## Request headers

|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|

> |Content-Type|application/json. Required.|

Reviewer 2: Usually, only requests that need a request body use the content-type header. Please, verify whether this DELETE operation really needs the content-type header. If that isn't the case, please update lines 44 and 68.

Author: removed.

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `204 No Content` response code.

## Examples

### Request

The following is an example of a request.

<!-- {
  "blockType": "request",
  "name": "update_certificateauthorityasentity"
}
-->
``` http

> DELETE https://graph.microsoft.com/beta/directory/certificateAuthorities/certificateBasedApplicationConfigurations/{certificateBasedApplicationConfigurations-Id}/trustedCertificateAuthorities/{trustedCertificateAuthorities-Id}

Reviewer 2: For clarity, please replace the placeholders, such as {certificateBasedApplicationConfigurations-Id}, with a real-looking ID. Examples with real-looking values help customers understand the API better. That's also the consistency bar for the Microsoft Graph API docs.

Please, apply this change to all API topics.

Author: updated.
